### ISSUE TITLE
[HOLD for payment 2022-11-09] [$250] Web - Profile - Changes made in offline mode are not saved

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#11632]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
Issue found when executing #11371

## Action Performed:
1. Log in any account
2. Navigate to Settings > Profile
3. Disable the internet connection in the device
4. Modify first name, last name, and your timezone
5. Save and verify everything looks like it updated correctly
6. Click back button to go back to settings page
7. Click profile and verify changes are preserved
8. Click back button to go back to settings page again
9. Go back online
10. Click on the Profile and check

## Expected Result:
Correct data should be still in your profile

## Actual Result:
Changes made in offline mode are not saved.
Correct data is shown only after refreshing the page.

## Workaround:
Refreshing the page fixes the issue
## Platform:

Where is this issue occurring?

- Web

**Version Number:** v1.2.12-0
**Reproducible in staging?:** Y
**Reproducible in production?:** new feature - save button disabled in offline in production
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/194371484-1054e723-c724-402b-86c3-73a24c7e628e.mp4

**Expensify/Expensify Issue URL:** 
**Issue reported by:** Applause internal team
**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by flodnv** [comment]
I am confused, does this not mean that https://github.com/Expensify/App/pull/11371 is broken @Julesssss ?

**Comment by Julesssss** [comment]
@flodnv it does, but I don't actually see the same result...

https://user-images.githubusercontent.com/10736861/194573739-89415103-d829-40ab-847d-65326757f93f.mov

Hi @kavimuru, perhaps there is an extra step that you did that I missed? Because I can't reproduce with those steps above (web)

**Comment by flodnv** [comment]
Ping @kavimuru ?

**Comment by kavimuru** [comment]
@Julesssss @flodnv Still able to reproduce. Edited the steps and adding a new video

https://user-images.githubusercontent.com/43996225/194971354-cc49e1a6-589f-41c8-ab32-b1a4cca67a7c.mp4

**Comment by Julesssss** [comment]
Okay, I also see this bug. Let's treat this as a separate issue as the steps are slightly different from the original issue.

**Comment by DaiTo141** [comment]
# Proposal

## Problem

- Currently, in offline mode when user edits profile and clicks save button, this action will add in `networkRequestQueue`, and will be called back if back online.
- When app reconnects, `api?command=ReconnectApp` will be called to refresh data and `api?command=UpdateProfile` to update new user profile asynchronous. So user data has not been updated in time on BE, `api?command=ReconnectApp` will return old user data.
Please check the evidences below:
![Screen_Shot_2022-10-12_at_17 27 13](https://user-images.githubusercontent.com/58692355/195332987-327cc025-3b43-4b9b-82de-7f85a8cb7bba.png)
![Screen Shot 2022-10-12 at 17 26 22](https://user-images.githubusercontent.com/58692355/195333039-09cddde3-08ed-4f56-ab82-497cedf6fd97.png)
## Solution

In https://github.com/Expensify/App/blob/299f07bfa29c0dd81a6126691aae29e161097baa/src/libs/actions/App.js#L119 we'll change `read` method to `write` method to add `api?command=ReconnectApp` to `SequentialQueue` to call synchronous

```diff
-  API.read('ReconnectApp', { policyIDListExcludingWorkspacesCreatedOffline }, {
+  API.write('ReconnectApp', { policyIDListExcludingWorkspacesCreatedOffline }, {
```


https://user-images.githubusercontent.com/58692355/195333665-afa544c9-7d4d-42eb-837b-6e9ce4eceab0.mp4

**Comment by tienifr** [comment]
# Proposal

## Problem

- Currently, in offline mode when user edits profile and clicks save button, this action will add in `networkRequestQueue`, and will be called back if back online.
- When app reconnects, `api?command=ReconnectApp` will be called to refresh data and `api?command=UpdateProfile` to update new user profile asynchronous. So user data has not been updated in time on BE, `api?command=ReconnectApp` will return old user data.
Please check the evidences below:
![Screen Shot 2022-10-12 at 17 26 22](https://user-images.githubusercontent.com/113963320/195336185-6d205d58-5465-421c-9275-38d06a724301.png)

![Screen_Shot_2022-10-12_at_17 27 13](https://user-images.githubusercontent.com/113963320/195336173-0396e729-50fc-46a7-87cc-08071096d709.png)


## Solution

In https://github.com/Expensify/App/blob/299f07bfa29c0dd81a6126691aae29e161097baa/src/libs/actions/App.js#L119 we'll change `read` method to `write` method to add `api?command=ReconnectApp` to `SequentialQueue` to call synchronous

```diff
-  API.read('ReconnectApp', { policyIDListExcludingWorkspacesCreatedOffline }, {
+  API.write('ReconnectApp', { policyIDListExcludingWorkspacesCreatedOffline }, {
```

https://user-images.githubusercontent.com/113963320/195336254-9d12b87c-61ba-4f04-b19c-b521e6de298c.mp4

**Comment by Julesssss** [comment]
Awaiting a review from @thesahindia, dropping the priority to weekly

**Comment by thesahindia** [comment]
I wasn't able to consistently repro this but yeah the [proposal](https://github.com/Expensify/App/issues/11632#issuecomment-1276039594) looks good to me and is working correctly.

C+ reviewed 🎀👀🎀

cc: @Julesssss

**Comment by flodnv** [comment]
>Awaiting a review from @thesahindia, dropping the priority to weekly

👎 Reviews should be done at least on a daily basis (and ideally several times a day: the most important thing one can do is unblock others)

**Comment by Julesssss** [comment]
I meant a proposal review, not a code review. But perhaps the same priority applies?

**Comment by flodnv** [comment]
Yes, as per the bot reapplying the daily label :)

**Comment by thesahindia** [comment]
> 👎 Reviews should be done at least on a daily basis (and ideally several times a day: the most important thing one can do is unblock others)

Thanks, will remember next time.

**Comment by tienifr** [comment]
@Julesssss Could you please send the Upwork link that I can apply to? thanks

**Comment by sakluger** [comment]
Upwork job posting: https://www.upwork.com/jobs/~011b81a6da237bb2ea

**Comment by tienifr** [comment]
Thanks @sakluger, I applied to the job and the PR will be ready by October 21

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by Julesssss** [comment]
Does Melvin think this was a regression? It wasn't

**Comment by mountiny** [comment]
@sakluger @Julesssss @thesahindia @tienifr This is still not fixed unfortunately https://github.com/Expensify/App/issues/12355 I have just tested on iOS staging and the old name comes back after a while

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by mountiny** [comment]
Confirmed this is now fixed in staging after latest deploy and we are good to pay the contributor out today @sakluger

**Comment by Julesssss** [comment]
Bumping for payment (to remove the outdated label)

**Comment by Julesssss** [comment]
Hey @sakluger, can you please pay out, or close the issue if we have already paid? Thanks

**Comment by thesahindia** [comment]
@sakluger, payment is pending.

**Comment by trjExpensify** [comment]
👋 Sorry for the delay on our side. Stepping in to process this to completion today. @thesahindia @tienifr offers have been sent to you both.

**Comment by thesahindia** [comment]
Accepted, thanks!

**Comment by trjExpensify** [comment]
Perfect, paid you both out. Thanks again! 🚀
